js/components/portal-dashboard/feedback/feedback-question-rows.tsx

```diff
@@ -22,6 +23,17 @@ interface IProps {
 export const FeedbackQuestionRows: React.FC<IProps> = (props) => {
   const { answers, feedbacks, currentActivity, currentStudentId, students, updateQuestionFeedback } = props;
 
+  const getFeedbackIcon = (feedback: string, feedbackData: Map<string, any>, answer: Map<string, any>) => {
+    let feedbackBadge = <AwaitingFeedbackQuestionBadgeIcon />;
+    if (feedback !== "") {
+      feedbackBadge = feedbackValidForAnswer(feedbackData, answer)
+                      ? <GivenFeedbackQuestionBadgeIcon />
+                      : <UpdateFeedbackQuestionBadgeIcon />;
+    }
```

Comment:
I probably would have used two ternary operators, but I think this is clearer!  All this functional programming has tried to beat the if/else out of me!